v2/cf-nodejs-logging-support/src/lib/config/jwt-service.ts

```diff
@@ -7,33 +8,36 @@ const DEFAULT_DYN_LOG_LEVEL_HEADER = "SAP-LOG-LEVEL";
 
 export class JWTService {
     // Read dyn. log level header name from environment var
-    private static headerName = process.env[ENV_DYN_LOG_HEADER];
-    private static dynLogLevelHeader = JWTService.headerName ? JWTService.headerName : DEFAULT_DYN_LOG_LEVEL_HEADER;;
+    private static instance: JWTService;
+    private headerName = process.env[ENV_DYN_LOG_HEADER];
+    private dynLogLevelHeader = this.headerName ? this.headerName : DEFAULT_DYN_LOG_LEVEL_HEADER;
 
-    constructor() {
+    private constructor() { }
 
+    public static getInstance(): JWTService {
+        if (!JWTService.instance) {
+            JWTService.instance = new JWTService();
+        }
+
+        return JWTService.instance;
     }
 
-    static getDynLogLevelHeaderName() {
+    getDynLogLevelHeaderName() {
         return this.dynLogLevelHeader;
     }
 
     setDynLogLevelHeader() {
-        JWTService.dynLogLevelHeader = JWTService.headerName ? JWTService.headerName : DEFAULT_DYN_LOG_LEVEL_HEADER;
+        this.dynLogLevelHeader = this.headerName ? this.headerName : DEFAULT_DYN_LOG_LEVEL_HEADER;
     }
 
-    // Binds the Loglevel extracted from JWT token to the given request logger
-    static bindDynLogLevel(token: string, logger: Logger) {
+    getDynLogLevel(token: string): string {
```

Comment:
I think it would make sense to use `string | null` as return type to make clear that null is a valid result.